src/Cohesity.Powershell/Scripts/DownloadFile/Download-CohesityFilesAndFolders.ps1

```diff
@@ -0,0 +1,106 @@
+        $rfObj = Invoke-RestApi -Method 'Get' -Uri $searchUrl -Headers $headers
+        
+        $entityInfo = $null
+        if ($rfObj.files) {
+            $entityInfo = $rfObj.files | Where-Object {$_.protectionSource.name -eq $ServerName}
+
+            if ($entityInfo) {
+                $AbsolutePath = $entityInfo[0].filename
+                $EncodedFileName = [System.Web.HttpUtility]::UrlEncode($AbsolutePath)
+                $SourceId = $entityInfo[0].sourceId
+                $JobId = $entityInfo[0].jobId
+                $ClusterId = $entityInfo[0].jobUid.clusterId
+                $ClusterIncarnationId = $entityInfo[0].jobUid.clusterIncarnationId
+                $IsDirectory = $entityInfo[0].isFolder
+                $ViewBoxId = $entityInfo[0].viewBoxId
+
+                # Get the information about snapshots that contain the specified file or folder. 
+                #   In addition, information about the file or folder is provided.
+                $queryParam = [ordered]@{}
+                $queryParam.Add('jobId', $JobId)
+                $queryParam.Add('clusterId', $clusterId)
+                $queryParam.Add('clusterIncarnationId', $ClusterIncarnationId)
+                $queryParam.Add('sourceId', $SourceId)
+                $queryParam.Add('filename', $EncodedFileName)
+                $queryString = '?' + ($queryParam.Keys.ForEach({"$_=$($queryParam.$_)"}) -join '&')
+
+                $snapshotUrl = $server + '/irisservices/api/v1/public/restore/files/snapshotsInformation' + $queryString
+                $snapshotObj = Invoke-RestApi -Method 'Get' -Uri $snapshotUrl -Headers $headers
+
+                # Get the required information for downloading the file
+                $latestSnapshot = $snapshotObj[0]
+                $dwldParam = [ordered]@{}
+                $dwldParam.Add('attemptNum', $latestSnapshot.snapshot.attemptNumber)
+                $dwldParam.Add('clusterId', $clusterId)
+                $dwldParam.Add('clusterIncarnationId', $ClusterIncarnationId)
+                $dwldParam.Add('entityId', $SourceId)
+                $dwldParam.Add('filepath', $EncodedFileName)
+                $dwldParam.Add('jobId', $JobId)
+                $dwldParam.Add('jobInstanceId', $latestSnapshot.snapshot.jobRunId)
+                $dwldParam.Add('jobStartTimeUsecs', $latestSnapshot.snapshot.startedTimeUsecs)
+                $dwldParam.Add('viewBoxId', $ViewBoxId)
+
+                $dwldQS = '?' + ($dwldParam.Keys.ForEach({"$_=$($dwldParam.$_)"}) -join '&')
+                $downloadUrl = $server + '/irisservices/api/v1/downloadfiles' + $dwldQS
+
+                # Perform web request to download the file from the cluster
+                $fn = Split-Path -Path $AbsolutePath -Leaf
+                $OutFile = Join-Path $(if($OutFile){$OutFile} else {$Home}) $fn
+                $resp = Invoke-RestApi -Method 'Get' -Uri $downloadUrl -Headers $headers -OutFile "$OutFile"
+
+                if ($resp) {
```

Comment:
For a successful execution add something like 
if($resp) {
#print message
$resp
}